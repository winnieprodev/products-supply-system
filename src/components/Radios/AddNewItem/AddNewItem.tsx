import { PlusIcon } from "components/Icons";

import * as Styled from "./addNewItem.styled";

interface OwnProps {
  onClick: () => void;
}

const AddNewItem: React.FC<React.PropsWithChildren<OwnProps>> = ({
  onClick,
  children,
}) => {
  return (
    <Styled.Block onClick={onClick}>
      <Styled.Icon>
        <PlusIcon />
      </Styled.Icon>
      <Styled.Content>{children}</Styled.Content>
    </Styled.Block>
  );
};

export default AddNewItem;
