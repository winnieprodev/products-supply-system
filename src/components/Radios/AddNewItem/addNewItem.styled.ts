import styled from "styled-components";
import {
  secondaryColor,
  secondaryColorDarker,
  secondaryColorLighter,
} from "styles/colors";

export const Block = styled.div`
  display: flex;
  flex-grow: 1;
  width: 100px;
  max-width: 33.3%;
  padding: 12px;
  align-items: center;
  cursor: pointer;
  background: white;
  border: 1px dashed ${secondaryColor};
  border-radius: 10px;
  transition: background ease 0.3s;

  &:hover {
    background-color: ${secondaryColorLighter};
  }
  &:active {
    background-color: ${secondaryColorDarker};
  }
`;

export const Icon = styled.div`
  width: 16px;
  height: 16px;
  color: #ccc;
  margin-right: 8px;
`;

export const Content = styled.div`
  color: #808080;
`;
