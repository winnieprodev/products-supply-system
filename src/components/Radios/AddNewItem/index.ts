export { default as AddNewItem } from "./AddNewItem";
