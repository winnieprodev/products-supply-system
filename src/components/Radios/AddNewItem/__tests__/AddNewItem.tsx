import { render } from "@testing-library/react";

import AddNewItem from "../AddNewItem";

describe("AddNewItem", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <AddNewItem onClick={jest.fn()}>test</AddNewItem>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
