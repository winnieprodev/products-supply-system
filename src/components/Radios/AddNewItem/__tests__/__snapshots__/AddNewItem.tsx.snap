// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewItem Snapshot tests it matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100px;
  max-width: 33.3%;
  padding: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background: white;
  border: 1px dashed #EEEEEE;
  border-radius: 10px;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
}

.c0:hover {
  background-color: #F3F3F3;
}

.c0:active {
  background-color: #E7E7E7;
}

.c1 {
  width: 16px;
  height: 16px;
  color: #ccc;
  margin-right: 8px;
}

.c2 {
  color: #808080;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-w-14 "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    test
  </div>
</div>
`;
